booking-app/src/main/java/org/motechproject/bookingapp/service/impl/VisitScheduleServiceImpl.java

```diff
@@ -115,6 +157,25 @@ public void savePlannedVisitDates(String subjectId, LocalDate primeVaccinationDa
         visitBookingDetailsService.createOrUpdate(calculatePlannedDates(subject.getVisits(), primeVaccinationDate).values());
     }
 
+    private void validateDate(String subjectId, LocalDate date) {
```

Comment:
you can pass screening date as the argument, so this request to db is not needed